File: tests/fixtures/repo-nestjs/src/report.controller.ts
import { Controller, Post, Body } from '@nestjs/common';

@Controller('report')
export class ReportController {
  @Post()
  async create(@Body() body) {
    await this.db.query('select * from users where id = ' + body.id);
    return { ok: true };
  }
}

File: tests/fixtures/repo-rails/app/controllers/report_controller.rb
# ガードが無い側（「ガード検出なし」と出るべき）
class ReportController < ApplicationController
  def create
    UserMailer.report(params[:email]).deliver_now
    render json: { ok: true }
  end
  def raw
    # 危険な関数（3 節）
    User.connection.execute("select * from users where id = #{params[:id]}")
  end
end
